packages/cli/internal/pkg/cli/logs_engine.go

```diff
@@ -45,24 +79,65 @@ func (o *logsEngineOpts) Execute() error {
 	}
 
 	logGroupName := contextInfo.EngineLogGroupName
-	if o.tail {
-		err = o.followLogGroup(logGroupName)
+	log.Debug().Msgf("Engine log group name: '%s'", logGroupName)
+
+	if o.workflowRunId == "" {
+		err = executeGetEngineLogForWholeGroup(o, logGroupName)
 	} else {
-		err = o.displayLogGroup(logGroupName, o.startTime, o.endTime, o.filter)
+		err = executeGetEngineLogForRunId(o, logGroupName)
 	}
 
 	return err
 }
 
+func executeGetEngineLogForWholeGroup(o *logsEngineOpts, logGroupName string) error {
+	if o.tail {
+		return o.followLogGroup(logGroupName)
+	}
+	return o.displayLogGroup(logGroupName, o.startTime, o.endTime, o.filter)
+}
+
+func executeGetEngineLogForRunId(o *logsEngineOpts, logGroupName string) error {
+	log.Info().Msgf("Getting log stream for workflow run '%s'", o.workflowRunId)
+
+	workflowRunLog, err := o.workflowManager.GetEngineLogByRunId(o.workflowRunId)
+	if err != nil {
+		return err
+	}
+
+	logStreamName := streamNameFromRunLog(workflowRunLog)
+	log.Debug().Msgf("Log stream name is: '%s'", logStreamName)
+
+	if logStreamName != "" {
+		if o.tail {
+			return o.followLogStreams(logGroupName, logStreamName)
+		}
+		return o.displayLogStreams(logGroupName, o.startTime, o.endTime, o.filter, logStreamName)
+	}
+
+	workflowStatus := workflowRunLog.WorkflowStatus
+	log.Warn().Msgf("Cannot find an engine log stream for workflow run '%s', the current status of the run is: '%s', "+
+		"the log will not be available until after the workflow is RUNNING", o.workflowRunId, workflowStatus)
+	return nil
+}
+
+func streamNameFromRunLog(workflowRunLog workflow.EngineLog) string {
```

Comment:
I think it should be a list of streams. It is possible we have stderr and stdout. I think assumption that we always have only one or another may result in sub-optimal user experience, when not all logs will be presented.